watchmode.go

```diff
@@ -0,0 +1,132 @@
+package templ
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"runtime"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+)
+
+var developmentMode = os.Getenv("TEMPL_DEV_MODE") == "true"
+
```

Comment:
As a new runtime function, do we want to place this in the `runtime` package?

Reply:
Good point. I wanted to reuse the existing functions and memory so that using templ 0.2.x generated code and 0.3.x generated code wouldn't cache the values twice, but given the plan to deprecate having runtime stuff in the templ package, I'll make the move.